File: app/src/main/java/com/example/admin/smarttranslator/Activities/FlagChangeActivity.java
package com.example.admin.smarttranslator.Activities;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearSnapHelper;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.SnapHelper;
import android.view.View;
import android.widget.TextView;

import com.example.admin.smarttranslator.Adapters.FlagAdapter;
import com.example.admin.smarttranslator.Entities.User;
import com.example.admin.smarttranslator.LayoutManagers.CenterZoomLayoutManager;
import com.example.admin.smarttranslator.R;

public class FlagChangeActivity extends AppCompatActivity {

    private  TextView from;
    private TextView to;
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.flag_change_layout);

        from = findViewById(R.id.from_lan);
        to = findViewById(R.id.to_lan);

        RecyclerView flagView = findViewById(R.id.list_flag);
        RecyclerView.LayoutManager layoutManager = new CenterZoomLayoutManager(this, 1, false, from);
        flagView.setLayoutManager(layoutManager);
        SnapHelper snapHelper = new LinearSnapHelper();
        snapHelper.attachToRecyclerView(flagView);
        RecyclerView.Adapter adapter = new FlagAdapter(User.getCurrentFlags(), this);
        flagView.setAdapter(adapter);

        RecyclerView flagView2 = findViewById(R.id.list_flag2);
        RecyclerView.LayoutManager layoutManager2 = new CenterZoomLayoutManager(this, 1, false, to);
        flagView2.setLayoutManager(layoutManager2);
        SnapHelper snapHelper2 = new LinearSnapHelper();
        snapHelper2.attachToRecyclerView(flagView2);
        RecyclerView.Adapter adapter2 = new FlagAdapter(User.getCurrentFlags(),this);
        flagView2.setAdapter(adapter2);

        flagView.postDelayed(() -> flagView.smoothScrollToPosition(7), 0);
        flagView2.postDelayed(() -> flagView2.smoothScrollToPosition(7),0);
    }

    public void onClickBackButton(View view) {
        Intent intent = new Intent();

        intent.putExtra("from", String.valueOf(from.getText()).toLowerCase());
        intent.putExtra("to", String.valueOf(to.getText()).toLowerCase());

        setResult(Activity.RESULT_OK, intent);
        finish();
    }
}

File: app/src/main/java/com/example/admin/smart_translator/app/FlagChangeActivity.kt
package com.example.admin.smart_translator.app

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.view.View
import android.widget.TextView

import com.example.admin.smart_translator.Adapters.FlagAdapter
import com.example.admin.smart_translator.Models.User
import com.example.admin.smart_translator.LayoutManagers.CenterZoomLayoutManager
import com.example.admin.smart_translator.R

class FlagChangeActivity : AppCompatActivity() {

    private var from: TextView? = null
    private var to: TextView? = null
    override fun onCreate(@Nullable savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.flag_change_layout)

        from = findViewById(R.id.from_lan)
        to = findViewById(R.id.to_lan)

        val flagView = findViewById<androidx.recyclerview.widget.RecyclerView>(R.id.list_flag)
        val layoutManager = CenterZoomLayoutManager(this, 1, false, from)
        flagView.layoutManager = layoutManager
        val snapHelper = androidx.recyclerview.widget.LinearSnapHelper()
        snapHelper.attachToRecyclerView(flagView)
        val adapter = FlagAdapter(User.currentFlags, this)
        flagView.adapter = adapter

        val flagView2 = findViewById<androidx.recyclerview.widget.RecyclerView>(R.id.list_flag2)
        val layoutManager2 = CenterZoomLayoutManager(this, 1, false, to)
        flagView2.layoutManager = layoutManager2
        val snapHelper2 = androidx.recyclerview.widget.LinearSnapHelper()
        snapHelper2.attachToRecyclerView(flagView2)
        val adapter2 = FlagAdapter(User.currentFlags, this)
        flagView2.adapter = adapter2

        flagView.postDelayed({ flagView.smoothScrollToPosition(7) }, 0)
        flagView2.postDelayed({ flagView2.smoothScrollToPosition(7) }, 0)
    }

    fun onClickBackButton(view: View) {
        val intent = Intent()

        intent.putExtra("from", from!!.text.toString().toLowerCase())
        intent.putExtra("to", to!!.text.toString().toLowerCase())

        setResult(Activity.RESULT_OK, intent)
        finish()
    }
}
